evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -70,13 +70,10 @@ def transform(self, X, y=None):
         Returns:
             pd.DataFrame: Transformed X
         """
-        X = _convert_to_woodwork_structure(X)
-        X = _convert_woodwork_types_wrapper(X.to_dataframe())
-        # Convert None to np.nan, since None cannot be properly handled
-        X = X.fillna(value=np.nan)
-
+        ww_X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(ww_X.to_dataframe())
         # Return early since bool dtype doesn't support nans and sklearn errors if all cols are bool
-        if (X.dtypes == bool).all():
+        if set(ww_X.logical_types.values()) == {ww.logical_types.Boolean} and X.isnull().sum().sum() == 0:
```

Comment:
@angela97lin do we need to change this to use woodwork? If I remember right, we had to put this patch in place because the sklearn simple imputer misbehaves when all cols have pandas dtype bool, right? In that case I'd argue this should remain as a pandas dtype check.

If the new code here is functionally equivalent to the old code, fine either way, just thought I'd ask.